google-cloud-tools-plugin/testSrc/com/google/cloud/tools/intellij/project/ProjectLoaderTest.java

```diff
@@ -0,0 +1,221 @@
+
+  @Before
+  public void setUp() {
+    projectLoader = new ProjectLoader();
+    when(mockGoogleApiClientFactory.getCloudResourceManagerClient(any()))
+        .thenReturn(mockCloudResourceManager);
+    testProject1 = new Project();
+    testProject1.setName("test-project");
+    testProject1.setProjectId("test-project-ID");
+    testProject2 = new Project();
+    testProject2.setName("test-project-2");
+    testProject2.setProjectId("test-project-2-ID");
+  }
+
+  @Test
+  public void loadUserProjects_invokesCallback() {
+    List<Project> projects = Arrays.asList(testProject1, testProject2);
+    mockListProjectsResponse(projects);
+
+    projectLoader.loadUserProjectsInBackground(mockUser, mockResultCallback);
+
+    verify(mockResultCallback).projectListReady(projects);
+  }
+
+  @Test
+  public void nullProjectList_invokes_callback_withEmptyList() {
+    mockListProjectsResponse(null);
+
+    projectLoader.loadUserProjectsInBackground(mockUser, mockResultCallback);
+
+    verify(mockResultCallback).projectListReady(Collections.emptyList());
+  }
+
+  @Test
+  public void loadUserProjects_sortsProjects_byName() {
+    Project aProject = new Project();
+    aProject.setName("a project");
+    aProject.setProjectId("The ID");
+    mockListProjectsResponse(Arrays.asList(testProject1, testProject2, aProject));
+
+    projectLoader.loadUserProjectsInBackground(mockUser, mockResultCallback);
+
+    verify(mockResultCallback)
+        .projectListReady(Arrays.asList(aProject, testProject1, testProject2));
+  }
+
+  @Test
+  public void deletedProjects_filteredFromResult() {
+    List<Project> projects = Arrays.asList(testProject1, testProject2);
+    testProject1.setLifecycleState("DELETE_REQUESTED");
+    mockListProjectsResponse(projects);
+
+    projectLoader.loadUserProjectsInBackground(mockUser, mockResultCallback);
+
+    List<Project> expectedList = Collections.singletonList(testProject2);
+    verify(mockResultCallback).projectListReady(expectedList);
+  }
+
+  @Test
+  public void projects_withEmptyId_filteredFromResult() {
+    List<Project> projects = Arrays.asList(testProject1, testProject2);
+    testProject1.setProjectId(null);
+    testProject2.setProjectId("");
+    mockListProjectsResponse(projects);
+
+    projectLoader.loadUserProjectsInBackground(mockUser, mockResultCallback);
+
+    verify(mockResultCallback).projectListReady(Collections.emptyList());
+  }
+
+  @Test
+  public void ioError_resultsIn_valid_errorMessage() throws IOException {
+    mockListProjectsResponse(null);
+    when(mockList.execute()).thenThrow(new IOException("IO issue"));
+
+    projectLoader.loadUserProjectsInBackground(mockUser, mockResultCallback);
+
+    verify(mockResultCallback)
+        .onError(GctBundle.getString("project.selector.loader.couldnotconnect"));
+  }
+
+  @Test
+  public void runtimeException_resultsIn_exceptionMessage() throws IOException {
+    // remove IDEA logging which invokes AssertionError.
```

Comment:
is this true for unit tests? I thought a different logger was implementation was loaded for test

Reply:
I checked again and it does stop the unit test when IntelliJ Logger.error() is called preventing the following logic to execute plus marking test as failed as a result of an error.